app/controllers/collections_controller.rb

```diff
@@ -189,4 +233,218 @@ def respond_with_error(message, status_code)
     end
   end
 
+  # Uploads a document given as json content
+  def upload_document_using_json(corpus_dir, file_basename, json_content)
+    absolute_filename = File.join(corpus_dir, file_basename)
+    Rails.logger.debug("Writing uploaded document contents to new file #{absolute_filename}")
+    create_file(absolute_filename, json_content)
+    absolute_filename
+  end
+
+  # Uploads a document given as a http multipart uploaded file or responds with an error if appropriate
+  def upload_document_using_multipart(corpus_dir, file_basename, file, collection_name)
+    absolute_filename = File.join(corpus_dir, file_basename)
+    if !file.is_a? ActionDispatch::Http::UploadedFile
+      raise ResponseError.new(412), "Error in file parameter."
+    elsif file.blank? or file.size == 0
+      raise ResponseError.new(412), "Uploaded file \"#{file_basename}\" is not present or empty."
+    else
+      Rails.logger.debug("Copying uploaded document file from #{file.tempfile} to #{absolute_filename}")
+      FileUtils.cp file.tempfile, absolute_filename
+      absolute_filename
+    end
+  end
+
+  # Uploads any documents in the item metadata and returns a copy of the item metadata with its metadata graph updated
+  def process_item_documents_and_update_graph(corpus_dir, item_metadata)
+    unless item_metadata["documents"].nil?
+      item_metadata["documents"].each do |document|
+        doc_abs_path = upload_document_using_json(corpus_dir, document["identifier"], document["content"])
+        unless doc_abs_path.nil?
+          item_metadata['metadata']['@graph'] = update_document_source_in_graph(item_metadata['metadata']['@graph'], document["identifier"], doc_abs_path)
+        end
+      end
+    end
+    item_metadata
+  end
+
+  # Updates the metadata graph for the given item
+  # Returns a copy of the item with the document sourcs in the graph updates to the path of an uploaded file when appropriate
+  def update_item_graph_with_uploaded_files(uploaded_files, item_metadata)
+    unless uploaded_files.nil?
```

Comment:
You don't need this wrapped in a `nil` check because your `process_uploaded_files` will never return a `nil`, it will return an empty array (which is good!)